compiler/optimizer/VPHandlers.cpp

```diff
@@ -1599,6 +1599,7 @@ TR::Node *constrainAload(OMR::ValuePropagation *vp, TR::Node *node)
                sym->isFinal()))
             haveClassLookaheadInfo = true;
 
+         bool allowForAOT = vp->comp()->getOption(TR_UseSymbolValidationManager);
```

Comment:
Given only classes are actually covered properly by the symbol validation manager (methods still use the "old" approach)...have you convinced yourself that the enhanced knowledge about classes doesn't end up enabling an incorrect runtime state due to the methods derived from those classes? Or will that be protected because the queries on the resulting method won't let things get out of hand?

Reply:
From what I've seen, `TR_MethodPointer` is only generated when dealing with Profiled Pointers, so it _should_ be ok using the "old" approach due to the class chain + first class loaded from classloader validations. Additionally, `TR_RamMethod` seems to be used only (or more specifically, can *only* be used) for the method being compiled, so that too should be ok using the "old" approach. That said, I'm doing some runs now with these changes made to the appropriate places
```
-               if (symbolKind == TR::SymbolType::typeClass && cg()->comp()->getOption(TR_UseSymbolValidationManager))
+               if (cg()->comp()->getOption(TR_UseSymbolValidationManager))
```
since I can't think of any good reason why the "new" approach shouldn't working, Towards the end of this work, I did find a bug in this code that I fixed, so perhaps I won't see that crash now.